tickit/devices/eiger/eiger_adapters.py

```diff
@@ -0,0 +1,264 @@
+
+        Args:
+            request (web.Request): The request object that takes the request method.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        await self.device.arm()
+
+        LOGGER.debug("Arming Eiger...")
+        return web.json_response(serialize(SequenceComplete(2)))
+
+    @HTTPEndpoint.put(f"/{DETECTOR_API}" + "/command/disarm")
+    async def disarm_eiger(self, request: web.Request) -> web.Response:
+        """A HTTP Endpoint for the 'disarm' command of the Eiger.
+
+        Args:
+            request (web.Request): The request object that takes the request method.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        await self.device.disarm()
+
+        LOGGER.debug("Disarming Eiger...")
+        return web.json_response(serialize(SequenceComplete(3)))
+
+    @HTTPEndpoint.put(f"/{DETECTOR_API}" + "/command/trigger")
+    async def trigger_eiger(self, request: web.Request) -> web.Response:
+        """A HTTP Endpoint for the 'trigger' command of the Eiger.
+
+        Args:
+            request (web.Request): The request object that takes the request method.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        trigger_message = await self.device.trigger()
+        self.device._set_state(State.IDLE)
+
+        LOGGER.debug(trigger_message)
+        return web.json_response(serialize(SequenceComplete(4)))
+
+    @HTTPEndpoint.put(f"/{DETECTOR_API}" + "/command/cancel")
+    async def cancel_eiger(self, request: web.Request) -> web.Response:
+        """A HTTP Endpoint for the 'cancel' command of the Eiger.
+
+        Args:
+            request (web.Request): The request object that takes the request method.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        await self.device.cancel()
+
+        LOGGER.debug("Cancelling Eiger...")
+        return web.json_response(serialize(SequenceComplete(5)))
+
+    @HTTPEndpoint.put(f"/{DETECTOR_API}" + "/command/abort")
+    async def abort_eiger(self, request: web.Request) -> web.Response:
+        """A HTTP Endpoint for the 'abort' command of the Eiger.
+
+        Args:
+            request (web.Request): The request object that takes the request method.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        await self.device.abort()
+
+        LOGGER.debug("Aborting Eiger...")
+        return web.json_response(serialize(SequenceComplete(6)))
+
+
+class EigerZMQAdapter(ZeroMQAdapter):
+    """An Eiger adapter which parses the data to send along a ZeroMQStream."""
+
+    device: EigerDevice
```

Comment:
Do we need an `EigerZMQAdapter`?

Reply:
I assumed we would based on a diagram @callumforrester drew a few weeks back on a whiteboard :thinking:

Reply:
Different adapters for different network interfaces seems nice, clean and separated to me

Reply:
I more meant because it doesn't extend `ZeroMQAdapter` in any way, seems superfluous

Reply:
Hmm, is there any device-specific logic in `ZeroMQAdapter` that should really be in here?

Reply:
Nah ZeroMQAdapter currently just sets up the zmq streams and outputs the response

Reply:
Ah, so the logic for getting data out of the eiger stream will come later?

Reply:
Yeah, was one of the things i was gonna do on the next branch